cmd/modelcmd/controller.go

```diff
@@ -287,8 +289,22 @@ func (w *sysCommandWrapper) Init(args []string) error {
 	}
 	if w.controllerName != "" {
 		if err := w.SetControllerName(w.controllerName); err != nil {
-			return errors.Trace(err)
+			return translateControllerError(w.ClientStore(), err)
 		}
 	}
-	return w.ControllerCommand.Init(args)
+	return w.ControllerCommand.Run(ctx)
+}
+
+func translateControllerError(store jujuclient.ClientStore, err error) error {
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	controllers, err := store.AllControllers()
```

Comment:
Nope, it reads your controllers.yaml. Disk access, while not ideal, still fairly reliable.